src/app/3d/qgs3dmapcanvaswidget.cpp

```diff
@@ -387,15 +387,16 @@ void Qgs3DMapCanvasWidget::resetView()
 
 void Qgs3DMapCanvasWidget::configure()
 {
-  QDialog dlg( this );
-  dlg.setWindowTitle( tr( "3D Configuration" ) );
-  dlg.setObjectName( QStringLiteral( "3DConfigurationDialog" ) );
-  dlg.setMinimumSize( 600, 460 );
-  QgsGui::enableAutoGeometryRestore( &dlg );
+  QDialog *dlg = new QDialog( this );
+  dlg->setAttribute( Qt::WA_DeleteOnClose );
+  dlg->setWindowTitle( tr( "3D Configuration" ) );
+  dlg->setObjectName( QStringLiteral( "3DConfigurationDialog" ) );
+  dlg->setMinimumSize( 600, 460 );
+  QgsGui::enableAutoGeometryRestore( dlg );
 
   Qgs3DMapSettings *map = mCanvas->map();
-  Qgs3DMapConfigWidget *w = new Qgs3DMapConfigWidget( map, mMainCanvas, mCanvas, &dlg );
-  QDialogButtonBox *buttons = new QDialogButtonBox( QDialogButtonBox::Ok | QDialogButtonBox::Apply | QDialogButtonBox::Cancel | QDialogButtonBox::Help, &dlg );
+  Qgs3DMapConfigWidget *w = new Qgs3DMapConfigWidget( map, mMainCanvas, mCanvas, dlg );
+  QDialogButtonBox *buttons = new QDialogButtonBox( QDialogButtonBox::Ok | QDialogButtonBox::Apply | QDialogButtonBox::Cancel | QDialogButtonBox::Help, dlg );
```

Comment:
Hmm if we make the dialog non-modal (which will be a good thing), I think it is maybe more appropriate to have just Close button instead of OK/Cancel which seem more appropriate for modal dialogs... what do you think?

Reply:
It looks like a good idea. I have updated the PR with this change.